movies/src/main/java/com/shotcutter/movies/movie/MovieService.java

```diff
@@ -0,0 +1,65 @@
+package com.shotcutter.movies.movie;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.data.mongodb.core.ReactiveMongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.data.mongodb.core.query.TextCriteria;
+import org.springframework.stereotype.Service;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class MovieService {
+
+    private final MovieRepository movieRepository;
+    private final ReactiveMongoTemplate mongoTemplate;
+
+    public MovieService(MovieRepository movieRepository,
+                        ReactiveMongoTemplate mongoTemplate) {
+        this.movieRepository = movieRepository;
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    public Mono<Movie> findById(Long id) {
+        return movieRepository.findById(id);
+    }
+
+    public Mono<Page<Movie>> search(Pageable pageable, String query, List<String> genres) {
```

Comment:
hmm, flux provides skip and take, I'd suggest to use Flux<Movie> instead of this